File: app/src/main/java/com/example/kursach/domain/repository/AuthRepository.kt
package com.example.kursach.domain.repository

import com.example.kursach.domain.model.UserProfile
import com.example.kursach.domain.usecase.RegistrationData

interface AuthRepository {
    suspend fun login(email: String, password: String): Result<UserProfile>
    suspend fun register(data: RegistrationData): Result<UserProfile>
    suspend fun fetchProfile(forceRefresh: Boolean = false): Result<UserProfile>
    suspend fun logout(): Result<Unit>
}

File: lib/src/domain/repositories/auth_repository.dart
import '../entities/user_profile.dart';
import '../usecases/registration_data.dart';
import '../usecases/profile_update_data.dart';

abstract class AuthRepository {
  Future<UserProfile> login({required String email, required String password});

  Future<UserProfile> register(RegistrationData data);

  Future<UserProfile> fetchProfile({bool forceRefresh = false});

  Future<void> logout();

  Future<UserProfile> updateProfile(ProfileUpdateData data);
}
